Lib/typing.py

```diff
@@ -1217,7 +1217,9 @@ def _determine_new_args(self, args):
         return tuple(new_args)
 
     def copy_with(self, args):
-        return self.__class__(self.__origin__, args, name=self._name, inst=self._inst)
+        return self.__class__(self.__origin__, args, name=self._name, inst=self._inst,
```

Comment:
I found the issue. The problem is that `_ConcatenateGenericAlias` inherits from `_GenericAlias`. That PR calls `super().copy_with` and passes everything up. Now if we look at `_ConcatenateGenericAlias`'s `__init__`:
https://github.com/python/cpython/blob/9510c9511a3596291523b671f8bc59f581e8f2fb/Lib/typing.py#L1425-L1427

It effectively expands to:
```python
 super().__init__(*args, _typevar_types=..., _paramspec_tvars=..., 
                  _typevar_types=(TypeVar, ParamSpec), 
                  _paramspec_tvars=True) 
```
Note the duplicates! That's why the error message says `TypeError: typing._GenericAlias.__init__() got multiple values for keyword argument '_typevar_types'`. The solution is to special-case `_ConcatenateGenericAlias`.
```suggestion
        if isinstance(self, _ConcatenateGenericAlias):
            return self.__class__(self.__origin__, args)
        return self.__class__(self.__origin__, args, name=self._name, inst=self._inst,
```

Reply:
Thanks for the investigation and for the proposal! This is actually quite depressing, though. I'm really sad if `_GenericAlias` now has to be aware of one of its child classes `_ConcatenateGenericAlias`, and carve out an exception in the base implementation. Doesn't that seem backwards? It may be the only practical option we have given the current state of things, but isn't this like a rookie OOD smell? Doesn't this further contribute to the spaghettification of `typing.py`?

Reply:
Actually, I take that back. This might be unavoidable? I think the "design" of `copy_with` itself is broken. It looks like it's trying to pull double- or triple-duty depending on who's using it and where. In that world, all OOD bets are off, so this might be making the best of a bad situation?

Or maybe `_ConcatenateGenericAlias.__init__(…, **kwargs, _typevar_types=(TypeVar, ParamSpec), _paramspec_tvars=True)` is the wrong way to go? Not sure. This seems like one hack on top of another. What's one more? 😬